pkg/client/rootd/dns/server.go

```diff
@@ -249,6 +256,17 @@ func (s *Server) resolveInCluster(c context.Context, q *dns.Question) (result dn
 	return result, rCode, nil
 }
 
+func (s *Server) ResolveAlias(q *dns.Question) *string {
+	for i := range s.config.Mappings {
+		mappingName := s.config.Mappings[i].Name + "."
+		if mappingName == q.Name || mappingName == strings.TrimSuffix(q.Name, s.clusterDomain) {
```

Comment:
```suggestion
	if len(s.config.Mappings) == 0 {
		return nil
	}
	qName := q.Name[:len(q.Name)-1] // strip trailing dot
	for i := range s.config.Mappings {
		if mappingName == qName  {
			dotName := s.config.Mappings[i].AliasFor + "."
			return &dotName
		}
		if strings.HasPrefix(q.Name, mappingName) && q.Name == mappingName + ".svc." + s.clusterDomain {
			dotName := s.config.Mappings[i].AliasFor + ".svc." + s.clusterDomain
			return &dotName
		}
```
Motivation:
- no repeated string concatenation and also no string concatenation unless `q.Name` is prefixed with `mappingName` (saves some CPU-cycles).
- if a search path ending with the cluster domain has been applied to `q.Name`, then it will most likely be `.svc.<cluster domain>`, not just `<cluster domain>`.

I'm not 100% convinced that a cluster domain test is necessary, but if such a test matches, then the returned CNAME must also have the `.svc.<cluster domain>` part appended.

Thinking more about this, the `.svc.<cluster domain>` check is in fact a synthetic mapping and as such, it can be added in advance to the mapping table so that no extra test is needed here. Such an addition could be conditioned to only happen on platforms where the resolver is subjected to search paths (Windows and Linux with systemd-resolved). I don't think it is ever needed when using the overriding resolver (Linux in docker uses that) or on macOS (where all namespaces become domain names).

Reply:
I'm not sure we talk about the same thing. Currently for this function [here](https://github.com/telepresenceio/telepresence/blob/afdf8eba6efeeb537bbc8f2c3b5dedd25e96b830/pkg/client/rootd/dns/server.go#L662),

```cli
curl echo-service.blue
```

```golang
func (s *Server) resolveQuery(q *dns.Question, dv *cacheEntry) (dnsproxy.RRs, int, error) {
    // Here, q.name = `echo-service.blue.`
}
```

In the other hand:

```cli
curl echo-service
```

```golang
func (s *Server) resolveQuery(q *dns.Question, dv *cacheEntry) (dnsproxy.RRs, int, error) {
    // Here, q.name = `echo-service.cluster.local.`
}
```

There is a default root domain when we only specify a name that we need to get ride of / ignore. 

Here the default DNS resolver on Mac:

```cli
➜  ~ cat /etc/resolver/telepresence.local
# Generated by telepresence

port 59553
domain cluster.local # Means that cluster.local is added by default.
nameserver 127.0.0.1
search cluster.local
```

Reply:
@knlambert can you please try and remove the `search` from the emitted `telepresence.local`  file? I'm pretty sure it's bogus.

Reply:
Pretty sure that It won't be there anymore :D that's what this field is supposed to do. The search this is only added in the telepresence.local file. The others don't have it, but sure let me try :)

Reply:
There are two places in the code trying to inject that line : 

https://github.com/telepresenceio/telepresence/blob/ae0bd806e2e95e04117de621bfe070c11f6a3ea6/pkg/client/rootd/dns/server_darwin.go#L201
https://github.com/telepresenceio/telepresence/blob/ae0bd806e2e95e04117de621bfe070c11f6a3ea6/pkg/client/rootd/dns/server_darwin.go#L194

If I remove the search line, `curl echo-easy` stops working. The requests never reach the local server though :(

Reply:
And you only remove it from the `telepresence.local` file? Not from any other generated file?

Reply:
Try this:
1. Create a successful intercept.
2. Verify that `curl echo-easy` works.
3. Manually edit the `telepresence.local` file and remove the `search` line.
4. Verify that `curl echo-easy` still works